test/SILGen/enum_debuginfo.swift

```diff
@@ -0,0 +1,7 @@
+// RUN: %target-swift-frontend -emit-silgen %S/Inputs/enum_debuginfo_other.swift -primary-file %s -module-name enum_debuginfo -g -Xllvm -sil-print-debuginfo | %FileCheck %s
+
+public func makeEnum() -> MyEnum {
+  return .hasPayload(argument: 123)
+}
+
+// CHECK-NOT: enum_debuginfo_other.swift
```

Comment:
Can you add any kind of positive check, too? Otherwise this will pass even if someone symlinks swiftc to /bin/true ...

Reply:
If I'm going to be super nit-picky about the test name, technically, this is a SILLocation, and pre-debuginfo :-)